raiden_contracts/data/source/services/OneToN.sol

```diff
@@ -95,13 +96,83 @@ contract OneToN is Utils {
             assert(expiration_block > 0);
             emit Claimed(sender, receiver, expiration_block, transferable);
 
-            // event SessionSettled(_key, expiration_block);
             require(deposit_contract.transfer(sender, receiver, transferable), "deposit did not transfer");
         }
         return transferable;
     }
 
-    // TODO: gas saving function to claim multiple IOUs and free space in one transaction
+    /// @notice Submit multiple IOUs to claim the owed amount.
+    /// This is the same as calling `claim` multiple times, except for the reduced gas cost.
+    /// @param senders Addresses from which the amounts are transferred
+    /// @param receivers Addresses to which the amounts are transferred
+    /// @param amounts Owed amounts of tokens
+    /// @param expiration_blocks Tokens can only be claimed before this time
+    /// @param one_to_n_address Address of this contract
+    /// @param signatures Sender's signatures concatenated into a single bytes array
+    /// @return Amount of transferred tokens
+    function bulkClaim(
+        address[] memory senders,
+        address[] memory receivers,
+        uint256[] memory amounts,
+        uint256[] memory expiration_blocks,
+        address one_to_n_address,
+        bytes memory signatures
+    )
+        public
```

Comment:
Make it `external` and replace `memory` with `calldata` in the above lines.  Then we save some gas (I believe).